src/common/image.c

```diff
@@ -1380,7 +1380,7 @@ static uint32_t _image_import_internal(const int32_t film_id, const char *filena
     {
       GList *imgs = NULL;
       imgs = g_list_prepend(imgs, GINT_TO_POINTER(id));
-      DT_DEBUG_CONTROL_SIGNAL_RAISE(darktable.signals, DT_SIGNAL_GEOTAG_CHANGED, g_list_copy((GList *)imgs), 0);
```

Comment:
I'm not sure this is right. The list passed to the signal is deleted by the signal itself when passed to all connected routine. See signal.c:

```
  { "dt-geotag-changed", NULL, NULL, G_TYPE_NONE,  g_cclosure_marshal_generic, 2, geotag_arg,
    G_CALLBACK(_image_geotag_destroy_callback), FALSE }, // DT_SIGNAL_GEOTAG_CHANGED
```

Where we have a `_image_geotag_destroy_callback` recorded.

Reply:
This is correct -- the old code had a leak, because we've newly constructed a single-element list inside this block, and the pointer to it goes away right after the call to RAISE when we hit the end of the block.  So if we **do** copy the list, the copy is freed and the original leaks.
I'll make it more explicit by using prepend(NULL,...).

Reply:
I think, you are correct.